src/lib.rs

```diff
@@ -33,11 +35,29 @@ pub fn run(config: Config) {
     // that is passed to each of the commands
     {
         let mut data = client.data.write();
-        data.insert::<Config>(Arc::new(config));
+        data.insert::<Config>(Arc::new(config.clone()));
     }
 
     client.with_framework(
         StandardFramework::new()
+            .before(move |ctx, msg, cmd_name| {
+                if !is_authorized(&msg.author.id.to_string(), config.clone()) {
+                    if config.no_auth_commands.contains(cmd_name) {
```

Comment:
Storing the list of commands as a string without any separator could lead to more commands being mistakenly allowed. For example, if the `foo-bar` command can be used without authentication, `foo` would be too.